### DISCUSSION TITLE
Hot reload or flipper emulator ?

### DISCUSSION THREAD

**Discussion by MassivDash** [dcoles/flipperzero-rs#161] [Q&A] [answered]
Hello, thanks for great work on the official bump to v1 on flipper. Now that we can stabilise any app development for flipper using rust lang, I have a question on tips for app development.

Is there a better way to develop more graphic app on flipper than pushing the build into the device ? 

I am looking for something that will either emulate flipper or use flipper to run the app in hot reload mode.

Any tips would be greatly appreciated

**Comment by dcoles** [comment, marked as answer]
I'm not personally aware of a full Flipper Zero emulator, but a quick search did bring up the archived [flippulator](https://github.com/Milk-Cool/flippulator) project that allows compiling Flipper Zero apps for PC.

You can combine [`cargo-watch`](https://crates.io/crates/cargo-watch) and [`run-fap`](https://github.com/flipperzero-rs/flipperzero/blob/main/tools/src/bin/run-fap.rs) from the tools directory to automatically build and start the binary on a physical device.

An example can now be found in the [`flipperzero-template` README](https://github.com/flipperzero-rs/flipperzero-template?tab=readme-ov-file#build-and-run-on-change):

```sh
cargo watch -s 'cargo build --release && run-fap target/thumbv7em-none-eabihf/release/my-project.fap'
```
